lib/ngx/process.md

```diff
@@ -0,0 +1,195 @@
+Name
+====
+
+`ngx.process` - manage the nginx processes for OpenResty/ngx_lua.
+
+Table of Contents
+=================
+
+* [Name](#name)
+* [Status](#status)
+* [Synopsis](#synopsis)
+    * [Enable privilege process and get process type](#enable-privilege-process-and-get-process-type)
+* [Methods](#methods)
+    * [type](#type)
+    * [privilege](#privilege)
+* [Community](#community)
+    * [English Mailing List](#english-mailing-list)
+    * [Chinese Mailing List](#chinese-mailing-list)
+* [Bugs and Patches](#bugs-and-patches)
+* [Author](#author)
+* [Copyright and License](#copyright-and-license)
+* [See Also](#see-also)
+
+Status
+======
+
+This Lua module is currently considered experimental.
+
+Synopsis
+========
+
+Enable privilege process and get process type
+-----------------------------------------
+
+```nginx
+# http config
+init_by_lua_block {
+    local process = require "ngx.process"
+
+    -- enable privileged process
+    process.privilege(true)
+
+    -- output process type
+    ngx.log(ngx.INFO, "process type:", process.type())
+}
+
+init_worker_by_lua_block {
+    local process = require "ngx.process"
+    ngx.log(ngx.INFO, "process type:", process.type())
+}
+
+server {
+    # ...
+    location = /t {
+        content_by_lua_block {
+            local process = require "ngx.process"
+            ngx.say("process type:", process.type())
+        }
+    }
+}
+
+```
+
+The example config above produces an output to `error.log` when
+server starts:
+
+```
+[lua] init_by_lua:8: process type:0             --> master process
+[lua] init_worker_by_lua:3: process type:5      --> privilege process
+[lua] init_worker_by_lua:3: process type:3      --> worker process
```

Comment:
I think it's better to return string values like "master", "worker", "privileged agent", and "helper". Constants are difficult to handle for the caller.